src/extension/mapml/src/main/java/org/geoserver/mapml/MapMLHTMLGetFeatureOutputFormat.java

```diff
@@ -0,0 +1,195 @@
+            Operation getFeature)
+            throws IOException, ServiceException {
+        Request request = Dispatcher.REQUEST.get();
+        HttpServletRequest httpRequest = request.getHttpRequest();
+
+        List<FeatureCollection> featureCollections = featureCollectionResponse.getFeatures();
+        if (featureCollections.size() != 1) {
+            throw new ServiceException(
+                    "MapML OutputFormat does not support Multiple Feature Type output.");
+        }
+        FeatureCollection featureCollection = featureCollections.get(0);
+        if (!(featureCollection instanceof SimpleFeatureCollection)) {
+            throw new ServiceException("MapML OutputFormat does not support Complex Features.");
+        }
+
+        // What if there is a reprojection?
+        SimpleFeatureCollection fc = (SimpleFeatureCollection) featureCollection;
+        ReferencedEnvelope projectedBbox = extractBbox(fc);
+        LayerInfo layerInfo = gs.getCatalog().getLayerByName(fc.getSchema().getTypeName());
+        CoordinateReferenceSystem crs = projectedBbox.getCoordinateReferenceSystem();
+        ProjType projType = parseProjType(request);
+        double longitude;
+        double latitude;
+        ReferencedEnvelope geographicBox;
+        try {
+            geographicBox = projectedBbox.transform(DefaultGeographicCRS.WGS84, true);
+            longitude = geographicBox.centre().getX();
+            latitude = geographicBox.centre().getY();
+        } catch (TransformException | FactoryException e) {
+            throw new ServiceException("Unable to transform bbox to WGS84", e);
+        }
+
+        boolean flipCoords = CRS.getAxisOrder(crs).equals(CRS.AxisOrder.NORTH_EAST);
+        double lat = flipCoords ? longitude : latitude;
+        double lon = flipCoords ? latitude : longitude;
+
+        MapMLHTMLOutput output =
+                new MapMLHTMLOutput.HTMLOutputBuilder()
+                        .setLongitude(lon)
+                        .setLatitude(lat)
+                        .setRequest(httpRequest)
+                        .setLayerLabel(layerInfo.getTitle())
+                        .setProjType(projType)
+                        .setProjectedBbox(projectedBbox)
+                        .setSourceUrL(buildGetFeature(request))
+                        .build();
+
+        // write to output
+        OutputStreamWriter osw = new OutputStreamWriter(out, gs.getSettings().getCharset());
+        osw.write(output.toHTML());
+        osw.flush();
+    }
+
+    private ProjType parseProjType(Request request) throws ServiceException {
+        try {
+            Map<String, Object> rawKvp = request.getRawKvp();
+            String srs;
+            if (rawKvp.containsKey("SRSNAME")) {
+                srs = (String) rawKvp.get("SRSNAME");
+            } else if (rawKvp.containsKey("SRS")) {
+                srs = (String) rawKvp.get("SRS");
+            } else {
+                srs = "EPSG:4326";
```

Comment:
maybe we should default to OSMTILE?

Reply:
I gather that because OSMTILE does not include the poles, that Andrea prefers to use WGS84.  I can live with that. On the other hand, "OSMTILE" is the default map projection used by the \<mapml-viewer> if you don't specify the `projection` attribute.  That shouldn't impact GeoServer though, since as Andrea points out the OSMTILE can't cover all earthly data.  
`WGS84` could be the default here.

Reply:
Actual live mapml should be served as `text/mapml`.  The preview application is (now, in WMS) served as `text/html; subtype=mapml`.

Reply:
> Should we default to OSMTILE?

I should be more clear: I think we should default to WGS84 to be consistent with how it's done on the WMS side.

Reply:
Correct, we are defaulting to WGS84 indeed.